utils/message-generator/src/parser/actions.rs

```diff
@@ -52,7 +54,23 @@ impl Sv2ActionParser {
                 match result.get("type").unwrap().as_str().unwrap() {
                     "match_message_type" => {
                         let message_type = u8::from_str_radix(&result.get("value").unwrap().as_str().unwrap()[2..], 16).expect("Result message_type should be an hex value starting with 0x and not bigger than 0xff");
-                        action_results.push(ActionResult::MatchMessageType(message_type));
+                        match result.get("condition") {
+                            Some(condition_inner) => {
+                                let condition_ =
+                                    serde_json::from_value::<Condition>(condition_inner.clone())
+                                        .unwrap();
+                                match condition_ {
+                                    Condition::WaitUntil => {
+                                        action_results.push(ActionResult::MatchMessageType(
+                                            message_type,
+                                            Some(Condition::WaitUntil),
+                                        ))
+                                    }
+                                }
```

Comment:
```suggestion
                                match serde_json::from_value::<Condition>(condition_inner.clone()) {
                                    Ok(Condition::WaitUntil) => {
                                        action_results.push(ActionResult::MatchMessageType(
                                            message_type,
                                            Some(Condition::WaitUntil),
                                        ))
                                    },
                                    Err(e) => {
                                        panic!("Failed to deserialize condition: {}", e);
                                    }
                                }
```

Reply:
@lorbax do you agree with this suggestion?

if yes, you can simply clicl the `Commit suggestion` button above, no need to manually implement this on your branch

Reply:
I accept this suggestion but I prefer to force push because I want the commit history to look as it is (single commit). Do you have a particular reason for adding a commit with this small suggestion?

Reply:
> Do you have a particular reason for adding a commit with this small suggestion?

not really! feel free to force push from your local changes